plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,317 @@
+          a lua function that returns a boolean indicating the parser should be disabled.
+        '';
 
-      customCaptures = mkOption {
-        type = types.attrsOf types.str;
-        default = { };
-        description = "Custom capture group highlighting";
-      };
+      custom_captures = helpers.defaultNullOpts.mkAttrsOf types.str { } ''
+        Custom capture group highlighting.
+      '';
+    };
+
+    incremental_selection = {
+      enable = helpers.defaultNullOpts.mkBool false ''
+        Incremental selection based on the named nodes from the grammar.
+      '';
 
-      incrementalSelection =
+      keymaps =
         let
-          keymap =
-            default:
-            mkOption {
-              type = types.str;
-              inherit default;
-              description = "Key shortcut";
-            };
+          mkKeymap = default: helpers.defaultNullOpts.mkStr default "Key shortcut";
         in
         {
-          enable = mkEnableOption "incremental selection based on the named nodes from the grammar";
-          keymaps = {
-            initSelection = keymap "gnn";
-            nodeIncremental = keymap "grn";
-            scopeIncremental = keymap "grc";
-            nodeDecremental = keymap "grm";
-          };
+          init_selection = mkKeymap "gnn";
+          node_incremental = mkKeymap "grn";
+          scope_incremental = mkKeymap "grc";
+          node_decremental = mkKeymap "grm";
         };
+    };
 
-      indent = mkEnableOption "tree-sitter based indentation";
+    indent = {
+      enable = helpers.defaultNullOpts.mkBool false ''
+        Whether to enable tree-sitter based indentation.
+      '';
+    };
 
-      folding = mkEnableOption "tree-sitter based folding";
+    ensure_installed = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+      Either "all" or a list of languages to ensure installing.
+    '';
 
-      languageRegister = mkOption {
-        type = with types; attrsOf (either str (listOf str));
-        description = ''
-          This is a wrapping of the `vim.treesitter.language.register` function.
-          Register specific parsers to one or several filetypes.
-          The keys are the parser names and the values are either one or several filetypes.
-        '';
-        default = { };
-        example = {
-          cpp = "onelab";
-          python = [
-            "myFiletype"
-            "anotherFiletype"
-          ];
-        };
-      };
+    ignore_install = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+      List of parsers to ignore installing (for "all")
+    '';
+
+    parser_install_dir = helpers.defaultNullOpts.mkStr "$XDG_DATA_HOME/nvim/treesitter" ''
+      Location of the parsers to be installed by the plugin (only needed when `nixGrammars` is disabled).
+
+      This default might not work on your own install, please make sure that `$XDG_DATA_HOME` is set if you want to use the default. Otherwise, change it to something that will work for you!
```

Comment:
```suggestion
      This default might not work on your own install, please make sure that `$XDG_DATA_HOME` is set if you want to use the default.
      Otherwise, change it to something that will work for you!
```